jetty-websocket/websocket-javax-tests/src/test/java/org/eclipse/jetty/websocket/javax/tests/ServerConfigTest.java

```diff
@@ -0,0 +1,137 @@
+import javax.websocket.server.ServerEndpoint;
+
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.websocket.javax.server.config.JavaxWebSocketServletContainerInitializer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ServerConfigTest
+{
+    private Server server;
+    private WebSocketContainer client;
+    private ServerConnector connector;
+
+    private static final long idleTimeout = 500;
+    private static final int maxTextMessageSize = 50;
+    private static final int maxBinaryMessageSize = 60;
+    private static final long asyncSendTimeout = 200;
+
+    @BeforeEach
+    public void start() throws Exception
+    {
+        server = new Server();
+        connector = new ServerConnector(server);
+        server.addConnector(connector);
+
+        ServletContextHandler contextHandler = new ServletContextHandler(ServletContextHandler.SESSIONS);
+        contextHandler.setContextPath("/");
+        server.setHandler(contextHandler);
+
+        JavaxWebSocketServletContainerInitializer.configure(contextHandler, (context, container) ->
+        {
+            container.setDefaultMaxSessionIdleTimeout(idleTimeout);
+            container.setDefaultMaxTextMessageBufferSize(maxTextMessageSize);
+            container.setDefaultMaxBinaryMessageBufferSize(maxBinaryMessageSize);
+            container.setAsyncSendTimeout(asyncSendTimeout);
+            container.addEndpoint(ConfigTestSocket.class);
+            container.addEndpoint(AnnotatedOnMessageSocket.class);
+        });
+
+        server.start();
+        client = ContainerProvider.getWebSocketContainer();
+    }
+
+    @AfterEach
+    public void stop() throws Exception
+    {
+        server.stop();
+    }
+
+    @ServerEndpoint("/containerDefaults")
+    public static class ConfigTestSocket
+    {
+        @OnOpen
+        public void onOpen(Session session)
+        {
+            assertThat(session.getMaxIdleTimeout(), is(idleTimeout));
+            assertThat(session.getMaxTextMessageBufferSize(), is(maxTextMessageSize));
+            assertThat(session.getMaxBinaryMessageBufferSize(), is(maxBinaryMessageSize));
+            assertThat(session.getAsyncRemote().getSendTimeout(), is(asyncSendTimeout));
+        }
+    }
+
+    @ServerEndpoint("/annotatedOnMessage")
+    public static class AnnotatedOnMessageSocket
+    {
+        @OnOpen
+        public void onOpen(Session session)
+        {
+            assertThat(session.getMaxTextMessageBufferSize(), is(111));
+            assertThat(session.getMaxBinaryMessageBufferSize(), is(-1));
```

Comment:
This check currently fails.

This issue also came up in #2661.
The API has the default `maxMessageSize` value for the `@OnMessage` annotation to be `-1` which means infinite message size. We didn't want that to be the default as it would be too easy for a client to send a huge message and make the server blow up with an `OutOfMemoryError`. But we don't know if they have specifically set `maxMessageSize = -1` in the annotation or it is just using the default value.

Currently if they set `maxMessageSize = -2` or `maxMessageSize = 0` it will result in infinite message size but a value of `-1` will not override the container default value, in this case `60`.

Reply:
Right, so the check should be `session.getMaxBinaryMessageBufferSize() > 0`, correct?

Reply:
@sbordet still not sure what to do about this, in the spec `-1` is defined as no maximum message size, but if they specifically set `-1` in the `onOpen` annotation it does not have any effect.